storage/database/db_manager.go

```diff
@@ -1901,6 +1882,12 @@ func (dbm *databaseManager) WritePreimages(number uint64, preimages map[common.H
 	preimageHitCounter.Inc(int64(len(preimages)))
 }
 
+func (dbm *databaseManager) DeleteTrieNode(hash common.Hash) {
+	if err := dbm.getDatabase(StateTrieDB).Delete(TrieNodeKey(hash)); err != nil {
```

Comment:
1. Does not need a write lock?
2. The updated `WriteTrieNode` function stores the provided data in two databases, whereas deletion only impacts a single database. Is the consistency maintained in this process?

Reply:
1. I believe respective DB implementations have a write lock inside. (e.g. NewLevelDB relies on [syndtr/leveldb](https://github.com/syndtr/goleveldb/blob/master/leveldb/db_write.go#L331) and it has internal write lock). The `database.Database` interface states that the Delete method must be thread-safe  (https://github.com/klaytn/klaytn/blob/v1.10.2/storage/database/interface.go#L70)
2. I simply followed the behavior of `DeleteCode`, which is another use-case of StateTrieDB. Currently, DeleteTrieNode is only used in tests, where there is only one MemDB.

Reply:
Fair point.
However, I was pursuing another consistency with respect to WriteCode and DeleteCode, where we Put() to two different databases and Delete() to one database. I applied the same principle to WriteTrieNode and DeleteTrieNode.

Reply:
Got it. thanks.